gems/pending/VMwareWebService/MiqVimInventory.rb

```diff
@@ -1891,6 +1905,55 @@ def addStoragePodObj(spObj)
     addObjHash(:StoragePod, spObj)
   end
 
+  def pbmProfiles_locked
+    log_header = "MiqVimInventory.pbmProfiles_locked:"
+    raise "pbmProfiles_locked: cache lock not held" unless @cacheLock.sync_locked?
+    return @pbmProfiles if @pbmProfiles
+
+    $vim_log.info "#{log_header} loading pbmProfile cache for #{@connId}"
+    begin
+      @pbmProfiles = {}
+
+      @cacheLock.sync_lock(:EX) if (unlock = @cacheLock.sync_shared?)
```

Comment:
Is this lock meant to prevent concurrent update of the ```@pbmProfiles```?  If so, can we build a new local pbmProfiles and only obtain the lock to swap out the hash held by ```@pbmProfiles```? That way, we can reduce the locked period.

(P.S. Not very family with the ```Sync``` module -- can't find any usage document on this ```Sync_m``` module. Beg for any pointers to docs)